### ISSUE TITLE
amp-next-page: Accidentally bundling v0.js

### ISSUE THREAD

**Issue by estherkim** [ampproject/amphtml#22357]
go/ampe/CNKj3ovwxJ3LPw

First seen April 17, 2019

```
Error: failed to attach shadow document for %s https://www-popsugar-com.cdn.ampproject.org/c/s/www.popsugar.com/amp/amp-next-content?nid=44773191&amp-next-page=1&native_boost=1&utm_campaign=native_redirect&utm_medium=amp&utm_source=tressugar: undefined is not an object (evaluating 'c.ownerDocument')
at ownerDocument (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:1282)
at contains_ (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:697)
at child (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:731)
at (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:865)
at (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:786)
at isRootLayer (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:334)
at declareLayer_ (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:151)
at ampdoc (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:1325)
at s (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service.js:401)
at holder (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service.js:166)
at ampdoc (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/layers-impl.js:1324)
at ampdoc (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/viewport/viewport-impl.js:166)
at ampdoc (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/viewport/viewport-impl.js:1311)
at s (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service.js:401)
at holder (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service.js:166)
at ampdoc (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/runtime.js:125)
at ampdoc (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/runtime.js:530)
at hostElement (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/runtime.js:621)
at shadowRoot (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/extensions/amp-next-page/0.1/next-page-service.js:215)
at shadowRoot (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/extensions/amp-next-page/0.1/next-page-service.js:303)
at (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/resources-impl.js:1026)
at state (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/vsync-impl.js:458)
at tasks (https://raw.githubusercontent.com/ampproject/amphtml/1905140117570/src/service/vsync-impl.js:415)
```

**Comment by estherkim** [comment]
@jridgewell can you see if this is an error we should track?

**Comment by prateekbh** [comment]
wierd but i dont see any error on the page itself

https://www-popsugar-com.cdn.ampproject.org/c/s/www.popsugar.com/amp/amp-next-content?nid=44773191&amp-next-page=1&native_boost=1&utm_campaign=native_redirect&utm_medium=amp&utm_source=tressugar

**Comment by jridgewell** [comment]
This is because amp-next-page.js includes **all** of v0.js's code, including it's own copy of `Resources` and `Layers`. /cc @peterjosling

**Comment by choumx** [comment]
@jridgewell Is this issue still outstanding?

**Comment by jridgewell** [comment]
Yah, it's gonna be a **big** change to fix this.

**Comment by wassgha** [comment]
Closed by #25629 for `amp-next-page` v0.1, also used by v2 in #25636
